client/src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -317,10 +320,17 @@ private synchronized PlatformError startTunnel(
   @Nullable
   private synchronized DetailedJsonError stopTunnel(@NonNull final String tunnelId) {
     if (!isTunnelActive(tunnelId)) {
-      return Errors.toDetailedJsonError(new PlatformError(
-          Platerrors.InternalError,
-          "VPN profile is not active"));
+        return Errors.toDetailedJsonError(new PlatformError(
+            Platerrors.InternalError,
+            "VPN profile is not active"));
+    }
+
+    // Stop reporting on the Client instance
+    if (this.outlineClient != null) {
+        this.outlineClient.stopReporting();
```

Comment:
Move to tearDownActiveTunnel. But I wonder if we should just implement that in stopRemoteDevice instead and revert this file. Why would we not align them?